### PR TITLE
compiler/tasking: fix compiler warning on tasking

### PR DESCRIPTION
**Pull request by anchao**
## Summary

compiler/tasking: fix compiler warning on tasking

```
ctc W549: ["serial/serial.c" 877/37] condition is always true
ctc W549: ["inode/fs_inodesearch.c" 72/8] condition is always true
ctc W545: ["obstack/lib_obstack_malloc.c" 69/1] missing 'return'
ctc W545: ["obstack/lib_obstack_malloc.c" 82/1] missing 'return'
ctc E246: ["map/mm_map.c" 67/41] left side of '.' or '->' is not struct or union
ctc E260: ["map/mm_map.c" 67/25] not an lvalue
ctc E246: ["map/mm_map.c" 80/3] left side of '.' or '->' is not struct or union
ctc E260: ["map/mm_map.c" 80/3] not an lvalue
```

## Impact

N/A

## Testing

Tricore / Aurix Tasking TC397

### CODE REVIEW

File: mm/map/mm_map.c [hunk 1]
```diff
@@ -64,7 +64,15 @@ static bool in_range(FAR const void *start, size_t length,
 
 int mm_map_lock(void)
 {
-  return nxrmutex_lock(&get_current_mm()->mm_map_mutex);
+  struct tcb_s *tcb = nxsched_self();
```
Comment (xiaoxiang781216) [inline mm/map/mm_map.c:67]: FAR

File: mm/map/mm_map.c [hunk 2]
```diff
@@ -77,7 +85,15 @@ int mm_map_lock(void)
 
 void mm_map_unlock(void)
 {
-  DEBUGVERIFY(nxrmutex_unlock(&get_current_mm()->mm_map_mutex));
+  struct tcb_s *tcb = nxsched_self();
```
Comment (xiaoxiang781216) [inline mm/map/mm_map.c:88]: FAR

File: mm/map/mm_map.c (hunk 1 above)
Comment (anchao) [reply mm/map/mm_map.c]: Done

File: mm/map/mm_map.c (hunk 2 above)
Comment (anchao) [reply mm/map/mm_map.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**